sys/net/routing/aodvv2/aodv.c

```diff
@@ -63,22 +64,20 @@ static mutex_t rerr_mutex;
 
 struct netaddr na_mcast;
 kernel_pid_t aodvv2_if_id;
-ipv6_addr_t aodvv2_prefix;
+ng_ipv6_addr_t aodvv2_prefix;
 int aodvv2_prefix_len;
 
-void aodv_init(void)
+void aodv_init(kernel_pid_t interface)
 {
     AODV_DEBUG("%s()\n", __func__);
 
-    /* init this thread's IPC msg queue (TODO: do I need to do this?) */
+    /* init this thread's IPC msg queue */
     msg_t msgq[RCV_MSG_Q_SIZE];
     msg_init_queue(msgq, sizeof msgq);
 
-    /* TODO: set & handle prefix and prefix_len properly (consider AODVV2_RIOT_PREFIXLEN!) */
-    aodvv2_prefix_len = 0;
-    /* TODO: set if_id properly (as param of aodv_init) */
-    aodvv2_if_id = 0;
-    net_if_set_src_address_mode(aodvv2_if_id, NET_IF_TRANS_ADDR_M_SHORT);
+    aodvv2_prefix_len = sizeof(ng_ipv6_addr_t);
```

Comment:
As @authmillenon stated usually our prefix length is 64.
However, this value is used here [1] (some lines below) registering the managed prefix to the FIB.
And in this case, the value just indicates how many bytes the used address type uses.
The actual bits for the managed prefix are derived when checking if the RP must be signalled for the given unknown destination [1].
I think the name of the variable is a bit confusing in this context.

[1] https://github.com/RIOT-OS/RIOT/pull/3445/files#diff-7b15e8fa8353d6ede3abe6c75c7b3e3cR131
[2] https://github.com/RIOT-OS/RIOT/blob/master/sys/net/network_layer/fib/fib.c#L318